Release/1.21.0

robkooper opened the pull request:
## 1.21.0 - 2022-08-23

**_Important:_** This update requires a MongoDB update schema due to a bug in the original migration  of showing summary statistics at the
space level. Make sure to start the application with -DMONGOUPDATE=1. You can also run the [fixCounts.js](https://github.com/clowder-framework/clowder/blob/develop/scripts/updates/fix-counts.js)
script prior to upgrading to minimize the downtime.

### Added
- api.Files jsonfile, adds two fields "downloads" and "views" [#228](https://github.com/clowder-framework/clowder/issues/228)
- Dataset and file scala.html pages incl schema.org jsonld metadata for (google)datasetsearch [#335](https://github.com/clowder-framework/clowder/issues/335)
- MiniUser and LicenseData now have to_jsonld methods to return string part of [#335](https://github.com/clowder-framework/clowder/issues/335) metadata
- LicenseData has urlViaAttributes used by it's to_jsonld to guess url when empty, for [#335](https://github.com/clowder-framework/clowder/issues/335)
- MRI previewer for NIFTY (.nii) files.
- Dataset page usually defaults to Files tab, but if no files will now show Metadata first
- HEIC (.heic) and HEIF (.heif) mimetypes to support new Apple iPhone image file format.
- In the docker container the folder /home/clowder/data is now whitelisted by default for uploading by reference. 
  This can be changed using the environment variable CLOWDER_SOURCEPATH.
- The current CLA for developers of clowder.

### Fixed
- Send email to all admins in a single email when a user submits 'Request access' for a space
- Send email to all admins and request user in a single email when any admin accepts/rejects 'Request access' for a space [#330](https://github.com/clowder-framework/clowder/issues/330)
- script/code to count space in files was not correct [#366](https://github.com/clowder-framework/clowder/issues/336)
- github actions would fail for docker builds due to secrets not existing
- Fix to remove dataset from a space [#349](https://github.com/clowder-framework/clowder/issues/349)

### Changed
- Utils.baseURL now on RequestHeader instead of Request[Any]
- MongoDB Service log error:'Not all dataset IDs found for Dataset|Folder bulk get request', now incl all the IDs notFound

ddey2 committed:
Fix to send notifications to all the admins in one email (#360)

* Fix to send notifications to all the admins in one email

Issue: 330

Signed-off-by: Dipannita Dey <ddey2@illinois.edu>

* Fixed recipients info on email while submit/accept/reject access to a space

Signed-off-by: Dipannita Dey <ddey2@illinois.edu>

Co-authored-by: Luigi Marini <lmarini@users.noreply.github.com>

cconsta1 committed:
Added an IFC previewer and a GLTF previewer (#340)

* New IFC previewer

* Added an IFC previewer and a GLTF previewer

* Changelog update

Co-authored-by: Chrysovalantis Constantinou <cconsta1@Chrysovalantiss-MacBook-Pro.local>
Co-authored-by: Luigi Marini <lmarini@users.noreply.github.com>

pgroves committed:
Adds MRI previewer for NIFTY (.nii) file format. (#241)

* Adds MRI previewer for NIFTY (.nii) file format.

* deleted vim swap files

* Added application/x-gzip to file types on which nii previewer runs on.

* Added changelog and contributors.

Co-authored-by: Luigi Marini <lmarini@illinois.edu>
Co-authored-by: Luigi Marini <lmarini@users.noreply.github.com>

robkooper committed:
CLA

robkooper committed:
fix status of spaces

original migration forgot to count files in folders

robkooper committed:
more mimetypes base on ext

robkooper committed:
whitelist of folders in docker

robkooper committed:
all changes

robkooper committed:
forgot to save filecount

robkooper committed:
check if secret is set (#371)

fixes #370

robkooper committed:
secrets can not be used in if

robkooper committed:
bump version

ddey2 committed:
Fix to remove a dataset from a space #349 (#373)

* Fix to remove a dataset from a space #349

Signed-off-by: Dipannita Dey <ddey2@illinois.edu>

robkooper committed:
Merge branch 'release/1.21.0' into misc-fixes

robkooper committed:
Merge pull request #367 from clowder-framework/misc-fixes

Misc fixes

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/clowder-framework/clowder?pullRequest=379) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/clowder-framework/clowder?pullRequest=379) before we can accept your contribution.<br/>**4** out of **5** committers have signed the CLA.<br/><br/>:white_check_mark: ddey2<br/>:white_check_mark: MBcode<br/>:white_check_mark: cconsta1<br/>:white_check_mark: robkooper<br/>:x: pgroves<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/clowder-framework/clowder?pullRequest=379) it.</sub>

robkooper committed:
fix changelog

MBcode committed:
Return folder as part of file metadata (#369)

* path: "/folder-1/folder-2"
* when not in folder make sure path has "/"
* api.Files jsonfile, adds two fields: downloads, views

MBcode committed:
file/dataset add schema.org (#332)

mapping file/dataset class to schema.org and return in file page
- when more that 10 items in list use "..." as final item

MBcode committed:
Not all dataset i ds found (#368)

list all ids of files that are in dataset but are not found.

robkooper committed:
Update CHANGELOG.md

MBcode committed:
Metadata tab select (#126)

* 1char fix to mongo-init

* added name to contributors

* permission change compiles

* select metadata if no files

* restore permission change

* if no hash test if can set metadata

* use tabsInURL.js

* split out hash part till can be tested

* split out url-hash till better tested

* tab- > space reformat

* same logic to get: tab-pane fade in active, too

* Dataset page usually defaults to Files tab, but if no files will now show Metadata first

* combine added

Co-authored-by: Luigi Marini <lmarini@users.noreply.github.com>

robkooper committed:
Merge remote-tracking branch 'origin/develop' into release/1.21.0

robkooper committed:
Update CHANGELOG.md
